### PR TITLE
added a check before running execution-callback

### PR DESCRIPTION
**Pull request by gurbuzali**
forward-port of https://github.com/hazelcast/hazelcast/pull/4628

### CODE REVIEW

**Comment by gurbuzali** [comment]
unrelated test fail
```
Test Result (2 failures / +2)

    com.hazelcast.map.query.QueryIndexingTest.testResultsHaveNullFields_whenUsingIndexes
    com.hazelcast.multimap.MultiMapTest.testListeners
```

**Comment by gurbuzali** [comment]
verify

File: hazelcast/src/test/java/com/hazelcast/executor/ExecutorServiceTest.java [hunk 1]
```diff
@@ -94,6 +95,31 @@ private IExecutorService createSingleNodeExecutorService(String name, int poolSi
...
+        String name = randomString();
+        IExecutorService executorService = instance2.getExecutorService(name);
+        BasicTestTask task = new BasicTestTask();
+        String key = generateKeyOwnedBy(instance1);
+        ICompletableFuture<String> future = (ICompletableFuture<String>) executorService.submitToKeyOwner(task, key);
+        final CountDownLatch latch = new CountDownLatch(1);
+        future.andThen(new ExecutionCallback<String>() {
+            @Override
+            public void onResponse(String response) {
+                latch.countDown();
```
Comment (eminn) [inline hazelcast/src/test/java/com/hazelcast/executor/ExecutorServiceTest.java]: One minor thing, we might add an assertion here to check whether response is correct or not like below.
 
`assertEquals(BasicTestTask.RESULT, response);`

**Comment by eminn** [comment]
It looks good to merge

**Pull request merged**